vsphere/resource_vsphere_host_virtual_switch.go

```diff
@@ -0,0 +1,135 @@
+package vsphere
+
+import (
+	"fmt"
+	"time"
+
+	"context"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/govmomi"
+)
+
+func resourceVSphereHostVirtualSwitch() *schema.Resource {
+	s := map[string]*schema.Schema{
+		"name": &schema.Schema{
+			Type:        schema.TypeString,
+			Description: "The name of the virtual switch.",
+			Required:    true,
+			ForceNew:    true,
+		},
+		"host": &schema.Schema{
+			Type:        schema.TypeString,
+			Description: "The name of the host to put this virtual switch on. This is ignored if connecting directly to ESXi, but required if not.",
+			Optional:    true,
+			ForceNew:    true,
+		},
+		"datacenter": &schema.Schema{
+			Type:        schema.TypeString,
+			Description: "The path to the datacenter the host is located in. This is ignored if connecting directly to ESXi. If not specified on vCenter, the default datacenter is used.",
+			Optional:    true,
+			ForceNew:    true,
+		},
+	}
+	mergeSchema(s, schemaHostVirtualSwitchSpec())
+
+	// Transform any necessary fields in the schema that need to be updated
+	// specifically for this resource.
+	s["active_nics"].Required = true
+	s["standby_nics"].Required = true
+
+	s["teaming_policy"].Default = "loadbalance_srcid"
+	s["check_beacon"].Default = false
+	s["notify_switches"].Default = true
+	s["failback"].Default = true
+
+	s["allow_promiscuous"].Default = false
+	s["forged_transmits"].Default = true
+	s["mac_changes"].Default = true
+
+	s["shaping_enabled"].Default = false
+
+	return &schema.Resource{
+		Create: resourceVSphereHostVirtualSwitchCreate,
+		Read:   resourceVSphereHostVirtualSwitchRead,
+		Update: resourceVSphereHostVirtualSwitchUpdate,
+		Delete: resourceVSphereHostVirtualSwitchDelete,
+		Schema: s,
+	}
+}
+
+func resourceVSphereHostVirtualSwitchCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*govmomi.Client)
+	host := d.Get("host").(string)
+	datacenter := d.Get("datacenter").(string)
+	ns, err := hostNetworkSystemFromName(client, host, datacenter)
+	if err != nil {
+		return fmt.Errorf("error loading network system: %s", err)
+	}
+
+	timeout := time.Duration(float64(d.Timeout(schema.TimeoutCreate)) * 0.8)
```

Comment:
as discussed on Slack - there's no guarantee a cancel would complete in `0.2 * timeout`?